lib/features/email/presentation/extensions/calendar_event_extension.dart

```diff
@@ -256,7 +256,60 @@ extension CalendarEventExtension on CalendarEvent {
     );
   }
 
+  String formatDate(date_format.DateLocale locale, DateTime dateTime) {
+    return date_format.formatDate(
+      dateTime,
+      [
+        date_format.DD,
+        ', ',
+        date_format.MM,
+        ' ',
+        date_format.dd,
+        ', ',
+        date_format.yyyy
+      ],
+      locale: locale
+    );
+  }
+
+  bool get isAllDayEvent {
+    if (startUtcDate != null && endUtcDate != null) {
+      final startDateValue = startUtcDate!.value;
+      final endDateValue = endUtcDate!.value;
+
+      final eventDurationInHours = startDateValue.difference(endDateValue).inHours;
+
+      final startHour = startDateValue.hour;
+      final startMinute = startDateValue.minute;
+      
+      final endHour = endDateValue.hour;
+      final endMinute = endDateValue.minute;
+      return startHour == 0 && startMinute == 0 && endHour == 0 && endMinute == 0 && eventDurationInHours % 24 == 0;
```

Comment:
IMO, we also need check `second` of start/end date

Reply:
done